IVRv2/app/services/websocket_service.py

```
@@ -0,0 +1,253 @@
+            task.cancel()
+        logger.info("Background processes cancelled")
+
+    async def _connect(self):
+        """Establish WebSocket connection with retry logic."""
+        while not self.is_connected:
+            try:
+                self._ws = await websockets.connect(self.connection_url)
+                self.is_connected = True
+                self.reconnect_attempts = 0
+                logger.info(f"Connected to WebSocket: {self.connection_url}")
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)